modules/casper.js

```diff
@@ -254,11 +287,47 @@ Casper.prototype.back = function back() {
  * @param  String  url     The url to download
  * @param  String  method  The method to use, optional: default GET
  * @param  String  data    The data to send, optional
+ * @param  Boolean async   Asynchroneous request? (default: false)
  * @return string          Base64 encoded result
  */
-Casper.prototype.base64encode = function base64encode(url, method, data) {
+Casper.prototype.base64encode = function base64encode(url, method, data, async) {
     "use strict";
-    return this.callUtils("getBase64", url, method, data);
+    async = ((typeof(async) !== "undefined")? async : false);
+	if ((phantom.version.major > 1) && (!async)) {
+	   throw new CasperError("async parameter must be 'true' with base64encode");
+    }        
+    var dl = {"url": url}, dls = this.ajax.filter( function searchDownload( ajax ){
+		   return (ajax.url === url) && (typeof (ajax.filename) === "undefined");
+		});	
+	if (dls.length === 0) {
+		this.ajax.push(dl);
+	} else {
+		dl = dls[0];
+	}    
+    dl.content = this.callUtils("getBase64", url, method, data, async);
```

Comment:
hmm I don't like the use of async as variable name, this could become a reserved keyword later...

how about ``asynchronous`` instead of ``async`` ?